### PR TITLE
Start block verification after sigchain cached is filled

### PR DESCRIPTION
**Pull request by yilunzhang**
This prevent new node from blocking consensus due to empty sigchain cache.

Signed-off-by: Yilun <zyl.skysniper@gmail.com>

### Proposed changes in this pull request
Explain the changes in this pull request in order to help the project maintainers understand the overall impact of it.

### Type (put an `x` where ever applicable)
- [x] Bug fix: Link to the issue
- [ ] Feature (Non-breaking change)
- [x] Feature (Breaking change)
- [ ] Documentation Improvement

### CODE REVIEW

File: consensus/consensus.go [hunk 1]
```diff
@@ -280,7 +274,7 @@ func (consensus *Consensus) saveAcceptedBlock(electedBlockHash common.Uint256) e
 
 	log.Infof("Accepted block height: %d, local ledger block height: %d, sync needed.", block.Header.UnsignedHeader.Height, chain.DefaultLedger.Store.GetHeight())
 
-	elc, loaded, err := consensus.loadOrCreateElection(heightToKey(block.Header.UnsignedHeader.Height))
+	elc, loaded, err := consensus.loadOrCreateElection(block.Header.UnsignedHeader.Height)
```
Comment (gdmmx) [inline consensus/consensus.go]: Why changed map's key from hash to height?
We should consider about elected in forked consensus, proposals from different blockChain maybe have same height.

Comment (yilunzhang) [reply consensus/consensus.go]: The key is not changed, heightToKey will be called inside function loadOrCreateElection

File: consensus/state.go [hunk 2]
```diff
@@ -31,6 +40,7 @@ func (consensus *Consensus) startGettingNeighborConsensusState() {
 			if localConsensusHeight == 0 || localConsensusHeight+1 < majorityConsensusHeight {
 				if majorityConsensusHeight+1 > localLedgerHeight {
 					consensus.setNextConsensusHeight(majorityConsensusHeight + 1)
+					consensus.localNode.SetMinVerifiableHeight(majorityConsensusHeight + 1 + por.SigChainMiningHeightOffset)
```
Comment (gdmmx) [inline consensus/state.go]: Could we make sure init node.minVerifiableHeight at here earlier than voting start?
As node.minVerifiableHeight uninitialized, it just only `0 + 1 + SigChainMiningHeightOffset = 5`

Comment (yilunzhang) [reply consensus/state.go]: It is initialized at the beginning of startGettingNeighborConsensusState()

**Comment by gdmmx** [comment]
LGTM

**Pull request merged**